docs/src/content/docs/components/file-tree.mdx

```diff
@@ -0,0 +1,237 @@
+---
+title: File Tree
+description: Learn how to display the structure of a directory with file icons and collapsible sub-directories.
+---
+
+import { FileTree } from '@astrojs/starlight/components';
+
+To display the structure of a directory with file icons and collapsible sub-directories, use the `<FileTree>` component.
+
+import Preview from '~/components/component-preview.astro';
+
+<Preview>
+
+<FileTree slot="preview">
+
+- astro.config.mjs an **important** file
+- package.json
+- README.md
+- src
+  - components
+    - **Header.astro**
+  - …
+- pages/
+
+</FileTree>
+
+</Preview>
+
+## Import
+
+```tsx
+import { FileTree } from '@astrojs/starlight/components';
+```
+
+## Usage
+
+Display a file tree with file icons and collapsible sub-directories using the `<FileTree>` component.
+
+Specify the structure of your files and directories with an [unordered Markdown list](https://www.markdownguide.org/basic-syntax/#unordered-lists) inside `<FileTree>`.
+Create a sub-directory using a nested list or add a `/` to the end of a list item to render it as a directory without specific content.
```

Comment:
We no longer show this `directory/` syntax in this simplified example, but it might make sense to bring it back given we don’t show it elsewhere?